internal/cmd/marketplace/marketplace_item.go

```diff
@@ -0,0 +1,377 @@
+// Copyright Mia srl
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package marketplace
+
+import (
+	"encoding/json"
+
+	"gopkg.in/yaml.v2"
+)
+
+func UnmarshalMarketplaceItem(data []byte) (Item, error) {
```

Comment:
is this interface used elsewhere in the codebase? I'd expect the item to only have one marshal function that may accept the "encoding" (json/yaml) as a input rather than having a different function for each encoding.

Also I wonder if it is actually needed; aren't the standard serializer enough? can't we extend the standard instead of wrapping them?

If we still want to wrap the standard fn I'd create a single one that accept the encoding instead of having two wrappers